pkg/executor/util/utils.go

```diff
@@ -84,18 +81,6 @@ func NewStandardStorageProvider(
 }
 
 func configureS3StorageProvider(cm *system.CleanupManager) (*s3.StorageProvider, error) {
-	dir, err := os.MkdirTemp(config.GetStoragePath(), "bacalhau-s3-input")
-	if err != nil {
-		return nil, err
-	}
-
-	cm.RegisterCallback(func() error {
-		if err := os.RemoveAll(dir); err != nil {
-			return fmt.Errorf("unable to clean up S3 storage directory: %w", err)
-		}
-		return nil
-	})
```

Comment:
So where does cleanup happen now?

Reply:
Other than tests, PrepareStorage is only ever called from ParallelPrepareStorage, and ParallelPrepareStorage is only used by the base executor (and the wasm executor) and each time returns a function that will call ParallelCleanup.  This is a step in the direction of us cleaning up the resources when they're no longer used so that storage is responsible for fetching the data, but the user is responsible for cleaning it up.